packages/compass-data-modeling/src/components/drawer/field-drawer-content.tsx

```diff
@@ -0,0 +1,229 @@
+  );
+
+  return {
+    isValid: !isDuplicate,
+    errorMessage: isDuplicate ? 'Field already exists.' : undefined,
+  };
+}
+
+const FieldDrawerContent: React.FunctionComponent<FieldDrawerContentProps> = ({
+  namespace,
+  fieldPath,
+  fieldPaths,
+  types,
+  relationships,
+  onCreateNewRelationshipClick,
+  onEditRelationshipClick,
+  onDeleteRelationshipClick,
+  onRenameField,
+  onChangeFieldType,
+}) => {
+  const { value: fieldName, ...nameInputProps } = useChangeOnBlur(
+    fieldPath[fieldPath.length - 1],
+    (fieldName) => {
+      const trimmedName = fieldName.trim();
+      if (trimmedName === fieldPath[fieldPath.length - 1]) {
+        return;
+      }
+      if (!isFieldNameValid) {
+        return;
+      }
+      onRenameField(namespace, fieldPath, [
+        ...fieldPath.slice(0, fieldPath.length - 1),
+        trimmedName,
+      ]);
+    }
+  );
+
+  const { isValid: isFieldNameValid, errorMessage: fieldNameEditErrorMessage } =
+    useMemo(
+      () => getIsFieldNameValid(fieldPath, fieldPaths, fieldName),
+      [fieldPath, fieldPaths, fieldName]
+    );
+
+  const handleTypeChange = (newTypes: string | string[]) => {
+    onChangeFieldType(namespace, fieldPath, types, newTypes);
+  };
+
+  return (
+    <>
+      <DMDrawerSection label="Field properties">
+        <DMFormFieldContainer>
+          <TextInput
+            label="Field name"
+            disabled={true} // TODO: enable when field renaming is implemented
+            data-testid="data-model-collection-drawer-name-input"
+            sizeVariant="small"
+            value={fieldName}
+            {...nameInputProps}
+            state={isFieldNameValid ? undefined : 'error'}
+            errorMessage={fieldNameEditErrorMessage}
+          />
+        </DMFormFieldContainer>
+
+        <DMFormFieldContainer>
+          <Combobox
+            data-testid="lg-combobox-datatype"
+            label="Datatype"
+            aria-label="Datatype"
+            disabled={true} // TODO(COMPASS-9659): enable when field type change is implemented
+            value={types}
+            size="small"
+            multiselect={true}
+            clearable={false}
+            onChange={handleTypeChange}
+          >
+            {BSON_TYPES.map((type) => (
+              <ComboboxOption key={type} value={type} />
+            ))}
+          </Combobox>
+        </DMFormFieldContainer>
+      </DMDrawerSection>
+
+      <RelationshipsSection
+        relationships={relationships}
+        emptyMessage="This field does not have any relationships yet."
+        getRelationshipLabel={([local, foreign]) => {
+          const labelField =
+            local.ns === namespace &&
+            JSON.stringify(local.fields) === JSON.stringify(fieldPath)
+              ? foreign
+              : local;
+          return [
+            labelField.ns ? toNS(labelField.ns).collection : '',
+            labelField.fields?.join('.'),
+          ].join('.');
```

Comment:
This hides relationship direction information from the list, feels like it's potentially confusing, are we sure we want that?

Reply:
I think the whole reason we went back on this is because it hides important information, this is why right now everything is packed into the "default" label, I don't think we should deivate from this honestly, at least until we start getting feedback about this